src/modules/user-auth/components/ForgetPasswordModal/ForgetPasswordModal.tsx

```diff
@@ -0,0 +1,33 @@
+import { VFC } from 'react';
+import { useModals } from '@ui/Modal/ModalContext';
+import Modal from '@ui/Modal/Modal';
+import ForgetPasswordForm from '../ForgetPasswordForm/ForgetPasswordForm';
+import { Icon } from '@iconify/react';
+import arrowIosBackFill from '@iconify/icons-eva/arrow-ios-back-fill';
+import { useAuth } from '@modules/user-auth/contexts/AuthContext';
+
+const ForgetPasswordModal: VFC = () => {
+  const { isAuthReady, isAuthenticated } = useAuth();
+  const { forgetPasswordModal, loginModal } = useModals();
+  return (
+    <Modal
+      size="md"
+      show={forgetPasswordModal.isShow}
+      onClose={
+        isAuthReady && !isAuthenticated
+          ? loginModal.show
+          : forgetPasswordModal.hide
+      }
+      header="Forget password?"
+      closeButtonIcon={
+        isAuthReady && !isAuthenticated ? (
```

Comment:
Is `isAuthReady && !isAuthenticated` in both cases the same? Better store them as a variable with a suitable name.

Reply:
On second though, `isAuthReady` is not necessary here... it is going to be false for a split second anyway